### PR TITLE
[PhpUnitBridge] fixed PHPUnit 8.3 compatibility: method handleError was renamed to __invoke

### PR DESCRIPTION
**Pull request by karser**
| Q             | A
| ------------- | ---
| Branch?       | 4.3 
| Bug fix?      | yes
| New feature?  | no 
| BC breaks?    | no     
| Deprecations? | no 
| Tests pass?   | yes    
| Fixed tickets | #32879   
| License       | MIT

The PHPUnit method  [handleError](https://github.com/sebastianbergmann/phpunit/blob/8.2.5/src/Util/ErrorHandler.php#L38) was renamed to [__invoke](https://github.com/sebastianbergmann/phpunit/blob/8.3/src/Util/ErrorHandler.php#L71) in v8.3.

So we should check in Symfony [DeprecationErrorHandler](https://github.com/symfony/symfony/blob/v4.3.3/src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php) if method `handleError` exists, otherwise call `__invoke`

It works with phpunit v8.2.5 and 8.3.2.
The PHPUnit handler is called when I trigger some error, e.g `iconv('fdsfs', 'fsdfds', '');`

### CODE REVIEW

File: src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php [hunk 1]
```diff
@@ -116,15 +115,38 @@ public static function collectDeprecations($outputFile)
...
     /**
      * @internal
      */
     public function handleError($type, $msg, $file, $line, $context = [])
     {
         if ((E_USER_DEPRECATED !== $type && E_DEPRECATED !== $type) || !$this->getConfiguration()->isEnabled()) {
-            $ErrorHandler = self::$utilPrefix.'ErrorHandler';
-
-            return $ErrorHandler::handleError($type, $msg, $file, $line, $context);
+            return static::callPhpUnitErrorHandler($type, $msg, $file, $line, $context);
```
Comment (karser) [inline src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php]: >We should then call the previous handler if one was set, or return false

@nicolas-grekas  Can we implement it here?
```
$result = static::callPhpUnitErrorHandler(....
if ($result !== false) {
    return $result;
}
```

**Review by nicolas-grekas — APPROVED** [review]
(I rebased this PR for 3.4 and made some fixes)

**Comment by karser** [comment]
I see. Anyways I'm glad I could help you with the research part and learnt some error handling stuff. Still have one question though, why do you return a `function that returns false` in getPhpUnitErrorHandler? isn't just `false` not enough in the set_error_handler callback?

**Comment by nicolas-grekas** [comment]
> why do you return a function that returns false in getPhpUnitErrorHandler? isn't just false not enough in the set_error_handler callback?

the return value is used with `call_user_func`, returning `false` here would then break.

**Comment by karser** [comment]
Should I apply fabbot patch?
image

**Comment by nicolas-grekas** [comment]
Nope, fabbot doesn't work for the bridge.

**Comment by nicolas-grekas** [comment]
Thank you @karser.

File: src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php [hunk 2]
```diff
@@ -44,6 +48,7 @@ public static function register($mode = 0)
         }
 
         $UtilPrefix = class_exists('PHPUnit_Util_ErrorHandler') ? 'PHPUnit_Util_' : 'PHPUnit\Util\\';
+        self::$isAtLeastPhpUnit83 = method_exists('PHPUnit\Util\ErrorHandler', '__invoke');
```
[the remark this answers is not in the document]
Comment (jderusse) [reply src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php]: nvm. Didn't notice that version were 3.4

File: src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php [hunk 3]
```diff
@@ -300,6 +299,29 @@ public static function collectDeprecations($outputFile)
         });
     }
 
+    /**
+     * @internal
+     */
+    public static function getPhpUnitErrorHandler()
+    {
+        if (!self::$isAtLeastPhpUnit83) {
+            return (class_exists('PHPUnit_Util_ErrorHandler', false) ? 'PHPUnit_Util_' : 'PHPUnit\Util\\').'ErrorHandler::handleError';
```
[the remark this answers is not in the document]
Comment (jderusse) [reply src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php]: nvm. Didn't notice that version were 3.4

**Pull request merged**